.github/workflows/init-release.yaml

```diff
@@ -0,0 +1,118 @@
+        required: true
+        type: string
+
+      TARGET_VERSION:
+        description: 'TARGET_VERSION to build manifests (e.g. 0.14.0-rc1) Note: the `v` prefix is not used'
+        required: true
+        type: string
+
+permissions: {}
+
+env:
+  TARGET_REMOTE: upstream
+  TARGET_TAG: "v${{ inputs.TARGET_VERSION }}"
+
+jobs:
+  prepare-release:
+    permissions:
+      contents: write  # for peter-evans/create-pull-request to create branch
+      pull-requests: write  # for peter-evans/create-pull-request to create a PR
+    name: Automatically generate version and manifests on ${{ inputs.TARGET_BRANCH }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@8410ad0602e1e429cee44a835ae9f77f654a6694  # v4.0.0
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ inputs.TARGET_BRANCH }}
+
+      - name: Check if TARGET_VERSION is well formed.
+        run: |
+          set -ue
+          # Target version must not contain 'v' prefix
+          if echo "${{ inputs.TARGET_VERSION }}" | grep -e '^v'; then
+            echo "::error::Target version '${{ inputs.TARGET_VERSION }}' should not begin with a 'v' prefix, refusing to continue." >&2
+            exit 1
+          fi
+          if ! echo "${{ inputs.TARGET_VERSION }}" | egrep -q '^[0-9]+\.[0-9]+\.[0-9]+$'; then
+            echo "Error: Target version '${{ inputs.TARGET_VERSION }}' is not well-formed. Must be X.Y.Z" >&2
+            exit 1
+          fi
+
+      - name: Checking for current git branch
+        run: |
+          set -ue
+          RELEASE_BRANCH=$(git rev-parse --abbrev-ref HEAD || true)
+          if [[ $RELEASE_BRANCH = release-* ]]; then
+            echo "***   branch is $RELEASE_BRANCH"
+          else
+            echo "Error: Branch $RELEASE_BRANCH is not release branch" >&2
+            exit 1
+          fi
+
+      - name: Checking for existence of git tag
+        run: |
+          set -ue
+          if git show-ref --tags "${{ env.TARGET_TAG }}" --quiet; then
+            echo "Error: Tag with version ${{ env.TARGET_TAG }} already exists." >&2
+            exit 1
+          fi
+
+      - name: Create VERSION information
+        run: |
+          set -ue
+          if ! test -f VERSION; then
+            echo "Error: You should be in repository root." >&2
+            exit 1
+          fi
+          echo "Bumping version from $(cat VERSION) to ${{ inputs.TARGET_VERSION }}"
+          echo "${{ inputs.TARGET_VERSION }}" > VERSION
+
+        # We install kustomize in the dist directory
+      - name: Add dist to PATH
+        run: |
+          echo "/home/runner/work/argocd-image-updater/argocd-image-updater/dist" >> $GITHUB_PATH
+
+      - name: Generate new set of manifests
+        run: |
+          set -ue
+          make manifests IMAGE_TAG="${{ env.TARGET_TAG }}"
+          git diff
```

Comment:
For this to work, we will need kustomize installed on the GitHub runner

Reply:
Added a step to install kustomize:

```yaml
      - name: Install Kustomize
        uses: imranismail/setup-kustomize@v2
        with:
          kustomize-version: '5.2.1'
```

At the beginning of next step, echo the kustomize version. In test run, it prints the correct version:
```
kustomize version v5.2.1
```